EventFilter/CTPPSRawToDigi/plugins/CTPPSPixelRawToDigi.cc

```diff
@@ -66,30 +80,59 @@ void CTPPSPixelRawToDigi::produce( edm::Event& ev,
 /// create product (digis & errors)
   auto collection = std::make_unique<edm::DetSetVector<CTPPSPixelDigi>>();
 
+  auto errorcollection = std::make_unique<edm::DetSetVector<CTPPSPixelDataError>>();
+  auto tkerror_detidcollection = std::make_unique<DetIdCollection>();
+  auto usererror_detidcollection = std::make_unique<DetIdCollection>();
 
   if(data_exist){
   es.get<CTPPSPixelDAQMappingRcd>().get( mapping);
 
-
     fedIds_   = mapping->fedIds();
 
     CTPPSPixelDataFormatter formatter(mapping->ROCMapping);
-    
-    bool errorsInEvent = false; 
+    formatter.setErrorStatus(includeErrors_);
+
+    bool errorsInEvent = false;
+    CTPPSPixelDataFormatter::DetErrors nodeterrors;
+
+
     for (auto aFed = fedIds_.begin(); aFed != fedIds_.end(); ++aFed) {
       int fedId = *aFed;
       
       edm::LogInfo("CTPPSPixelRawToDigi")<< " PRODUCE DIGI FOR FED: " <<  dec <<fedId << endl;
 
+      CTPPSPixelDataFormatter::Errors errors;
 /// get event data for this fed
       const FEDRawData& fedRawData = buffers->FEDData( fedId );
       
-      formatter.interpretRawData( errorsInEvent, fedId, fedRawData, *collection);
+      formatter.interpretRawData( errorsInEvent, fedId, fedRawData, *collection, errors);
+    
+      if(includeErrors_) {
+	typedef CTPPSPixelDataFormatter::Errors::iterator IE;
+	for (IE is = errors.begin(); is != errors.end(); is++) {
```

Comment:
this can be ```for( auto const& is : errors)``` and no need for the typedef in the line above

Reply:
done